x-pack/plugins/security_solution/public/common/components/exceptions/add_exception_flyout/index.tsx

```diff
@@ -166,7 +169,27 @@ export const AddExceptionFlyout = memo(function AddExceptionFlyout({
     }
   }, [jobs, ruleIndices]);
 
-  const [isIndexPatternLoading, { indexPatterns }] = useFetchIndex(memoRuleIndices);
+  const [isIndexPatternLoading, { indexPatterns: indexIndexPatterns }] =
+    useFetchIndex(memoRuleIndices);
+
+  const [indexPattern, setIndexPattern] = useState<DataViewBase>(indexIndexPatterns);
+
+  useEffect(() => {
+    if (ruleIndices != null && (dataViewId === '' || dataViewId == null)) {
+      setIndexPattern(indexIndexPatterns);
+    }
+  }, [dataViewId, indexIndexPatterns, ruleIndices]);
+
+  useEffect(() => {
```

Comment:
Going with your method of accessing the data view object through redux would be much better. This is an older comment and I know we talked about this in other meetings but just posting this comment for posterity - fetching the data view SO has to happen in a useEffect because of they async call.